backend/src/services/course/course.repo.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+/**
+ * Course Repository
+ *
+ * This module provides data access methods for course-related operations.
+ */
+
+/**
+ * Get all courses.
+ * @param {FastifyInstance} fastify - Fastify instance with Prisma client
+ * @returns {Promise<Array>} List of all courses
+ */
+async function getAllCourse(fastify) {
+  const courses = await fastify.db.courses.findMany();
+  return courses;
+}
+
+/**
+ * Get a course by ID.
+ * @param {FastifyInstance} fastify - Fastify instance with Prisma client
+ * @returns {Promise<Object>} Course object
+ */
+async function getCourseById(fastify, courseId) {
+  const course = await fastify.db.courses.findUnique({
+    where: { id: courseId },
+  });
+  return course;
+}
+
+/**
+ * Get users in a course.
+ * @param {FastifyInstance} fastify - Fastify instance with Prisma client
+ * @returns {Promise<Array>} List of users in the course
+ */
+async function getUsersInCourse(fastify, courseId) {
+  const users = await fastify.db.enrollments.findMany({
+    where: { course_id: courseId },
+  });
+  return users;
+}
+
+/**
+ * Get user details in a course.
+ * @param {FastifyInstance} fastify - Fastify instance with Prisma client
+ * @returns {Promise<Object>} User object
+ */
+async function getUserDetailsInCourse(fastify, courseId, userId) {
+  const user = await fastify.db.enrollments.findFirst({
+    where: {
+      user_id: userId,
+      course_id: courseId,
+    },
+  });
+  return user;
+}
```

Comment:
[Blocker]
These functions currently return the enrollment objects only. The function would also need to combine information from the users table for the users full profile with course context.